rockcraft/oci.py

```diff
@@ -355,19 +356,41 @@ def set_default_user(self, user: str) -> None:
         _config_image(image_path, params)
         emit.progress(f"Default user set to {user}")
 
-    def set_entrypoint(self) -> None:
-        """Set the OCI image entrypoint. It is always Pebble and CMD is null."""
+    def set_entrypoint(self, entrypoint_service: Optional[str]) -> None:
```

Comment:
I provided default to None for both set_entrypoint and set_cmd now.
As for `str | None`, switching to it would require carrying over the change in all similar places in the project, to keep the consistency. Therefore I am opting for keeping it as it is for the time being.

Reply:
@cjdcordeiro it's not the entrypoint that is None here, but the entrypoint_service. Or am I missing something?

Reply:
I didn't want to create a swirl here. I just meant that I kinda liked the syntax:

```
.set_entrypoint()
```

vs

```
.set_entrypoint(None)
```

by using default arguments. If this is not compliant to some internal rules you have (e.g. never provide default arguments), feel free to skip this. It was just a suggestion. 

The refactoring "Optional[str]" => " str | None", I agree it is out-of-scope as it touches a lot of places. It was also just a suggestion for your future development, provided by an outsider.

Both points do not block approval (which I have already given) nor require a re-review from my part!

Reply:
:man_facepalming: you're right...read too fast. It's fine for the `entrypoint-service` arg to default to None. Nevermind me :p